### PR TITLE
Adding Settings Registry Adapter feature

### PR DESCRIPTION
**Pull request by lumberyard-employee-dm**
## What does this PR do?

Integrates @noortor work for implementing a Settings Registry Adapter that can view and edit the associated Settings Registry.

The Settings Registry Adapter can be played in the [DPEDebugViewStandalone](https://github.com/o3de/o3de/blob/development/Code/Framework/AzQtComponents/CMakeLists.txt#L79-L91) application under the AzQtComponents folder.

The Settings Registry Adapter can modify existing fundamental type fields of objects and arrays which are string, bool, float and int, but currently cannot add or remove fields from an array or object.

Added the WM_DeleteOnClose attribute the [DPEDebugWindow class and DocumentPropertyEditor](https://github.com/o3de/o3de/blob/development/Code/Framework/AzQtComponents/AzQtComponents/DPEDebugViewStandalone/main.cpp#L217-L220) to fix shutdown crash in the DPEDebugViewStandalone application.

relates #2218

## How was this PR tested?

Manual testing of the Settings Registry Adapter in the DPEDebugViewStandalone application.
image

### CODE REVIEW

File: Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/SettingsRegistryAdapter.cpp [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+            };
+            AZ::SettingsRegistryInterface& settingsRegistry = m_originTracker->GetSettingsRegistry();
+            AZ::SettingsRegistryVisitorUtils::VisitField(settingsRegistry, visitorCallback, path);
+            m_builder.Label("");
+            SettingsRegistryDomData domData;
+            domData.m_settingsKeyPath = path;
+            domData.m_callback = [](AZStd::string_view, const AZ::Dom::Value&) -> AZ::Dom::Value
+            {
+                // When support for adding for objects/array types
+                // are implemented, This function will need to prompty the user
```
Comment (cgalvan) [inline Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/SettingsRegistryAdapter.cpp:268]:
```suggestion
                // are implemented, This function will need to prompt the user
```

File: Code/Framework/AzQtComponents/AzQtComponents/DPEDebugViewStandalone/main.cpp [hunk 2]
```diff
@@ -165,20 +166,11 @@ namespace AZ
...
-    class DPEDebugApplication : public AzToolsFramework::ToolsApplication
+    class DPEDebugApplication
+        : public AzToolsFramework::ToolsApplication
     {
     public:
-        DPEDebugApplication(int* argc = nullptr, char*** argv = nullptr)
-            : AzToolsFramework::ToolsApplication(argc, argv)
-        {
-            AZ::NameDictionary::Create();
-            AZ::AllocatorInstance<AZ::Dom::ValueAllocator>::Create();
```
Comment (cgalvan) [inline Code/Framework/AzQtComponents/AzQtComponents/DPEDebugViewStandalone/main.cpp:175]: Why don't we need this allocator instance anymore, or was it unused?

**Review by cgalvan — APPROVED** [review]

File: Code/Framework/AzQtComponents/AzQtComponents/DPEDebugViewStandalone/main.cpp (hunk 2 above)
Comment (lumberyard-employee-dm) [reply Code/Framework/AzQtComponents/AzQtComponents/DPEDebugViewStandalone/main.cpp:175]: This allocator gets created by the PropertyEditorSystem, which gets created when the PropertyManagerComponent is created as part of the ToolsApplication instance
The PropertyMangerComponent contains a [PropertyEditorToolsSystem](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyManagerComponent.h#L76-L78) instance for the Document Property Editor.
The PropertyEditorToolsSystem creates a [DocumentPropertyEditor::PropertyEditorSystem](https://github.com/o3de/o3de/blob/development/Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/PropertyEditorToolsSystem.h#L36-L37)

The PropertyEditorSystem then creates the allocator in it's [constructor](https://github.com/o3de/o3de/blob/development/Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/PropertyEditorSystem.cpp#L25-L31)

File: Code/Framework/AzCore/AzCore/Settings/SettingsRegistry.h [hunk 3]
```diff
@@ -96,6 +96,26 @@ namespace AZ
             Object
         };
 
+        //! Type of an integer
+        enum class Signedness
+        {
+            None,
```
Comment (spham-amzn) [inline Code/Framework/AzCore/AzCore/Settings/SettingsRegistry.h:102]: Does 'None' mean the type in a non-integer?

**Review by amzn-tmryan — APPROVED** [review]

**Pull request merged**